### PR TITLE
[traderepublic] New date format for dividend dates. Support for foreign currency. Corrected parsed amount (patch)

### LINKED ISSUE #487: Trade Republic - Es können keine Dividendenabrechnungen importiert werden
## Document
<details><summary>pb16080367223958130670726747825.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Obpmthdpw Ynvmxxu Vjaztezo",
    "Ohgvpwrhjvrltlo Vcpoßa 557",
    "I 53966 Omuryizh",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "389199716",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "vzbuwmo@cpdewfqvcehaq.fir",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ a1e1-7b14",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 18.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Applied Materials Inc.",
    "Registered Shares o.N.",
    "ISIN: US0382221051",
    "2 Stk.",
    "0,22 USD",
    "0,44 USD",
    "GESAMT",
    "0,44 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,07 USD",
    "Zwischensumme",
    "0,37 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "0,30 EUR",
    "GESAMT",
    "0,30 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "EA35920907292340905509",
    "2020-12-10",
    "0,30 EUR",
    "Applied Materials Inc. Registered Shares o.N. in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://app.tresor.one/p/5fce30fc0619f3003824d92f

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #488: Trade Republic. Es können keine Dividendenabrechnungen importiert werden
## Document
<details><summary>Abrechnung 2.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Skoxavksb Wüluz",
    "Wwtja Hkcrßh 8",
    "T 11599 Rtlnjtb",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "429045072",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "ujvewet@rssdapesrkxik.vpj",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ af9d-6028",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 09.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Intl Business Machines Corp.",
    "Registered Shares DL -,20",
    "ISIN: US4592001014",
    "6 Stk.",
    "1,63 USD",
    "9,78 USD",
    "GESAMT",
    "9,78 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-1,47 USD",
    "Zwischensumme",
    "8,31 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "6,85 EUR",
    "GESAMT",
    "6,85 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "LT98101274322366809252",
    "2020-12-10",
    "6,85 EUR",
    "Intl Business Machines Corp. Registered Shares DL -,20 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/3fd98d3f

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #489: [TRADE REPUBLIC] PDF wird nicht importiert [Problem: Datum?]
anscheinend hat TR das Datumsformat geändert, weshalb die PDF jetzt nicht mehr erkannt wird. Zumindest ist das das einzige, was mir als Unterschied zu älteren Abrechnungen, die funktionierten, auffällt. 

## Document
<details><summary>AbrechnungIBM12.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Dhwgybzsn Bgbwcj Vykwo",
    "Göxwvvraczzßr 542",
    "M 36134 Nvpayes",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "957822821",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "mmthzcr@djdxksciavubp.eeg",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ b315-2aa4",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 09.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Intl Business Machines Corp.",
    "Registered Shares DL -,20",
    "ISIN: US4592001014",
    "11,4984 Stk.",
    "1,63 USD",
    "18,74 USD",
    "GESAMT",
    "18,74 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-2,81 USD",
    "Zwischensumme",
    "15,93 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "13,13 EUR",
    "GESAMT",
    "13,13 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "DZ72745256080029961077",
    "2020-12-10",
    "13,13 EUR",
    "Intl Business Machines Corp. Registered Shares DL -,20 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/123f757e

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #492: Dividendenabrechnung Trade Republic
## Document
<details><summary>Abrechnung.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Jens Prangenberg",
    "Iöcvat Csheßg 764",
    "D 14224 Epgmdpl",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "140150901",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "qwkcmaa@kxcdacxvaxlps.ugk",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 5d1d-afd7",
    "AUSSCHÜTTUNG",
    "ÜBERSICHT",
    "Ausschüttung mit dem Ex-Tag 15.12.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "iSh.STOXX Europe 600 U.ETF DE",
    "Inhaber-Anteile",
    "ISIN: DE0002635307",
    "22,6484 Stk.",
    "0,133 EUR",
    "3,01 EUR",
    "GESAMT",
    "3,01 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "YH17693277548645065930",
    "2020-12-15",
    "3,01 EUR",
    "iSh.STOXX Europe 600 U.ETF DE Inhaber-Anteile in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/a6eb94c8

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #494: [Broker Trade Republic] Dividendenabrechnung kann nicht importiert werden
## Documents
<details><summary>pb16080371651578131113488428529.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Myrqoi Iswatk Zgkzemyw",
    "Gtwöwalhfugn 29",
    "M 29837 Ezdivguyfzaaz",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "424180502",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "pebkvkd@hqmhusnizejqi.xhs",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 8ku6-307f",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 09.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Intl Business Machines Corp.",
    "Registered Shares DL -,20",
    "ISIN: US4592001014",
    "2 Stk.",
    "1,63 USD",
    "3,26 USD",
    "GESAMT",
    "3,26 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,49 USD",
    "Zwischensumme",
    "2,77 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "2,29 EUR",
    "GESAMT",
    "2,29 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "CW38709448721526045098",
    "2020-12-10",
    "2,29 EUR",
    "Intl Business Machines Corp. Registered Shares DL -,20 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

<details><summary>pb16080400825748134030905865684.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Htibhj Otwdkg Jerrrpsk",
    "Jzqöahjglviq 14",
    "R 32584 Rqxizvwaappmu",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "122143726",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "xxwcxxy@upqeqlkkiexsb.nxw",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ k55f-16s6",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 24.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "S&P Global Inc.",
    "Registered Shares DL 1",
    "ISIN: US78409V1044",
    "3 Stk.",
    "0,67 USD",
    "2,01 USD",
    "GESAMT",
    "2,01 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,30 USD",
    "Zwischensumme",
    "1,71 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "1,41 EUR",
    "GESAMT",
    "1,41 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "TR07083491500850954797",
    "2020-12-10",
    "1,41 EUR",
    "S&P Global Inc. Registered Shares DL 1 in Wertpapierrechnung.",
    "Lagerland: USA",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/77656fe3

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #495: trade republik
## Documents
<details><summary>Abrechnung(16).pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstr. 42",
    "D 12345 Musterstadt",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "123456789",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "eqpswmt@bepainvgnzgbn.zyg",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 87af-bc43",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 10.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Exxon Mobil Corp.",
    "Registered Shares o.N.",
    "ISIN: US30231G1022",
    "25 Stk.",
    "0,87 USD",
    "21,75 USD",
    "GESAMT",
    "21,75 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-3,26 USD",
    "Zwischensumme",
    "18,49 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "15,25 EUR",
    "GESAMT",
    "15,25 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "RF58328550293189755278",
    "2020-12-10",
    "15,25 EUR",
    "Exxon Mobil Corp. Registered Shares o.N. in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

<details><summary>Abrechnung(15).pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstr. 42",
    "D 12345 Musterstadt",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "123456789",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "bjcpdfv@qepbozhidihcl.jvj",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 87af-bc43",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 10.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Exxon Mobil Corp.",
    "Registered Shares o.N.",
    "ISIN: US30231G1022",
    "25 Stk.",
    "0,87 USD",
    "21,75 USD",
    "GESAMT",
    "21,75 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-3,26 USD",
    "Zwischensumme",
    "18,49 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "15,25 EUR",
    "GESAMT",
    "15,25 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "VT97888868418447108823",
    "2020-12-10",
    "15,25 EUR",
    "Exxon Mobil Corp. Registered Shares o.N. in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

<details><summary>Abrechnung(14).pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstr. 42",
    "D 12345 Musterstadt",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "123456789",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "vgrextb@hafwbeablkzzo.ing",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 7377-e300",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 18.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Walgreens Boots Alliance Inc.",
    "Reg. Shares DL -,01",
    "ISIN: US9314271084",
    "57 Stk.",
    "0,468 USD",
    "26,65 USD",
    "GESAMT",
    "26,65 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-4,00 USD",
    "Zwischensumme",
    "22,65 USD",
    "Zwischensumme",
    "1,2153 EUR/USD",
    "18,64 EUR",
    "GESAMT",
    "18,64 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "CN45040759345656519727",
    "2020-12-11",
    "18,64 EUR",
    "Walgreens Boots Alliance Inc. Reg. Shares DL -,01 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

leonid_gurari@yahoo.de

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #496: Trade Republic Es können keine Dividendenabrechnungen importiert werden
## Document
<details><summary>Abrechnung.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstr. 42",
    "D 12345 Musterstadt",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "123456789",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "hdowbxz@daiewqqslbraj.rmc",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ e0e3-32a5",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 10.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Exxon Mobil Corp.",
    "Registered Shares o.N.",
    "ISIN: US30231G1022",
    "5 Stk.",
    "0,87 USD",
    "4,35 USD",
    "GESAMT",
    "4,35 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,65 USD",
    "Zwischensumme",
    "3,70 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "3,05 EUR",
    "GESAMT",
    "3,05 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "AY64531024235124999090",
    "2020-12-10",
    "3,05 EUR",
    "Exxon Mobil Corp. Registered Shares o.N. in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

## Community profile

Gunther Dilly guntherdilly2109@gmail.com

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #497: Trade Republic
## Document
<details><summary>Abrechnung 3.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Loxhm Zgarvm Vbemoe",
    "Dowzgsuqjßw 715",
    "V 24689 Njpyntj",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "857706826",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "qsyskrs@ialakhpjvfyue.nyq",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ a8a9-de26",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 18.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Walgreens Boots Alliance Inc.",
    "Reg. Shares DL -,01",
    "ISIN: US9314271084",
    "1 Stk.",
    "0,468 USD",
    "0,47 USD",
    "GESAMT",
    "0,47 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,07 USD",
    "Zwischensumme",
    "0,40 USD",
    "Zwischensumme",
    "1,2153 EUR/USD",
    "0,33 EUR",
    "GESAMT",
    "0,33 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "BH67064346387887274944",
    "2020-12-11",
    "0,33 EUR",
    "Walgreens Boots Alliance Inc. Reg. Shares DL -,01 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #498: Trade Republic dividenden Abrechnung nicht erkannt
## Document
<details><summary>IBMdividez_201215_195854.pdf</summary><p>

```json
[
  [
    "AUWGB YOWOPDQA QJYM BCXZ",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Max Mustermann",
    "Musterstr. 42",
    "D 12345 Musterstadt",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "123456789",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "vubgvnm@wxgyifzoawvat.elr",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 1720-afe5",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 09.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Intl Business Machines Corp.",
    "Registered Shares DL -,20",
    "ISIN: US4592001014",
    "0,2561 Stk.",
    "1,63 USD",
    "0,42 USD",
    "GESAMT",
    "0,42 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,06 USD",
    "Zwischensumme",
    "0,36 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "0,30 EUR",
    "Kapitalertragssteuer",
    "-0,04 EUR",
    "Solidaritätszuschlag",
    "-0,01 EUR",
    "GESAMT",
    "0,25 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "DG19005865627197054305",
    "2020-12-10",
    "0,25 EUR",
    "Intl Business Machines Corp. Registered Shares DL -,20 in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #503: [Trade Republic] - Dividende Exxon Mobil
## Document
<details><summary>pb16080383292868132277617713884.pdf</summary><p>

```json
[
  [
    "TRADE REPUBLIC BANK GMBH",
    "KASTANIENALLEE 32",
    "10435 BERLIN",
    "Erwjpnw Btiqlmbxxz",
    "Ohhrrdsdnkcßr 9G",
    "N 55924 Bfoyqpiynwda",
    "SEITE",
    "1",
    "von",
    "1",
    "DATUM",
    "15.12.2020",
    "DEPOT",
    "437529707",
    "Trade Republic Bank GmbH",
    "Kastanienallee 32",
    "10435 Berlin",
    "www.traderepublic.com",
    "zplvplt@itimdjbaimttx.ljt",
    "Sitz der Gesellschaft: Düsseldorf",
    "AG Düsseldorf HRB 85864",
    "USt-ID DE307510626",
    "Geschäftsführer",
    "Andreas Willius",
    "Karsten Müller",
    "ABRE",
    "/ 0468-de38",
    "DIVIDENDE",
    "ÜBERSICHT",
    "Dividende mit dem Ex-Tag 10.11.2020.",
    "POSITION",
    "ANZAHL",
    "ERTRÄGNIS",
    "BETRAG",
    "Exxon Mobil Corp.",
    "Registered Shares o.N.",
    "ISIN: US30231G1022",
    "4 Stk.",
    "0,87 USD",
    "3,48 USD",
    "GESAMT",
    "3,48 USD",
    "ABRECHNUNG",
    "POSITION",
    "BETRAG",
    "Quellensteuer DE für US-Emittent",
    "-0,52 USD",
    "Zwischensumme",
    "2,96 USD",
    "Zwischensumme",
    "1,2127 EUR/USD",
    "2,44 EUR",
    "GESAMT",
    "2,44 EUR",
    "BUCHUNG",
    "VERRECHNUNGSKONTO",
    "VALUTA",
    "BETRAG",
    "YA45142435946683552077",
    "2020-12-10",
    "2,44 EUR",
    "Exxon Mobil Corp. Registered Shares o.N. in Girosammelverwahrung.",
    "Diese Abrechnung wird maschinell erstellt und daher nicht unterschrieben.",
    "Sofern keine Umsatzsteuer ausgewiesen ist, handelt es sich gem. § 4 Nr. 8 UStG um eine umsatzsteuerfreie",
    "Leistung."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by SirGibihm**
Adepted the parser to support the new dividend format. 
* Fixes #503 (manually checked that the issue is the same as #498)
* Fixes #498 (Testfile added)
* Fixes #497 (Testfile added) 
* Fixes #496 (manually checked that the issue is the same as #498)
* Fixes #495 (manually checked that the issue is the same as #498)
* Fixes #494 (manually checked that the issue is the same as #498)
* Fixes #492 (manually checked that the issue is the same as #498)
* Fixes #489 (manually checked that the issue is the same as #498)
* Fixes #488 (manually checked that the issue is the same as #498)
* Fixes #487 (manually checked that the issue is the same as #498)
* Added support for foreign currency 
* Fixed issues where the amount was calculated incorrectly for transactions containing a witholding tax

### CODE REVIEW

**Comment by SirGibihm** [comment]
> 
> 
> I had done this to fix the bug:
> 
> ```
>   let [parsedDate, parsedDateTime] = createActivityDateTime(date, time);
> 
>   if (parsedDate === 'Invalid DateTime') {
>     [parsedDate, parsedDateTime] = createActivityDateTime(date, time, 'yyyy-MM-dd')
>   }
> ```
> 
> So basically, I tried multiple date formats.

Same with this fix, I check what date format to use by: 
```js
const dateFormat = findDateDividend(textArr).includes('-')
      ? 'yyyy-MM-dd'
      : 'dd.MM.yyyy';
```

File: src/brokers/traderepublic.js [hunk 1]
```diff
@@ -66,132 +66,93 @@ const findShares = textArr => {
...
-const findAndConvertNumber = (valueWithCurrency, textArr) => {
-  // Find a value and convert it to EUR using the rate given in the document.
-  const lineElements = valueWithCurrency.split(' ');
-  const value = Big(Math.abs(parseGermanNum(lineElements[0])));
-  if (lineElements[1] === 'EUR') {
-    // No currency conversion is required.
-    return value;
+const findForeignInformation = textArr => {
+  const foreignIdx = textArr.findIndex(line =>
+    /[1-9]+[0-9]*(,[0-9]*)? [A-Z]{3}\/[A-Z]{3}/.test(line)
```
Comment (dredav) [inline src/brokers/traderepublic.js:81]: What did the regex test? I think it is: `1,2127 EUR/USD`. Could you add a comment?

Comment (dredav) [reply src/brokers/traderepublic.js:81]: Starting with `[1-9]+` requires a fx rate >= 1.xx. But rates can also be possible with 0,42.

Comment (dredav) [reply src/brokers/traderepublic.js:81]: Proposal: `/[0-9]+(,[0-9]*)?\s[A-Z]{3}\/[A-Z]{3}/`

File: src/brokers/traderepublic.js [hunk 2]
```diff
@@ -332,6 +243,55 @@ const parseOverviewStatement = content => {
...
+    broker: 'traderepublic',
+    isin: findISIN(textArr),
+    company: findCompany(textArr),
+    shares: findShares(textArr),
+    tax: findTax(textArr, fxRate),
+    fee: findFee(textArr, fxRate),
+  };
+  if (isBuySingle(textArr) || isBuySavingsPlan(textArr)) {
+    activity.type = 'Buy';
+    const date = isBuySavingsPlan(textArr)
```
Comment (dredav) [inline src/brokers/traderepublic.js:259]: Should we handle the new date case also for buy and sell? Currently we don't have test data (my saving plans are executed tomorrow) but better safe than sorry?

File: src/brokers/traderepublic.js (hunk 1 above)
Comment (SirGibihm) [reply src/brokers/traderepublic.js:81]: Really good catch, fixed

File: src/brokers/traderepublic.js (hunk 2 above)
Comment (SirGibihm) [reply src/brokers/traderepublic.js:260]: I would not do that tbh. maybe its a dividend only thing and we did not receive notifcations about Buy and Sell operations as of yet.

**Review by dredav — APPROVED** [review]

**Comment by dredav** [comment]
Trade republic also changed the valuta format th saving splan executions **BUT** we don't use the field. We're using the the date from the line `Sparplanausführung am 16.12.2020 an der Lang & Schwarz Exchange.`. So parsing is possible!

**Pull request merged**